src/stackmap_checker/utils.c

```diff
@@ -20,9 +20,11 @@ char* get_binary_path()
     return buff;
 }
 
-Elf64_Ehdr* get_elf_header(const char *bin_name)
+Elf64_Ehdr* get_elf_header()
 {
+    char *bin_name = get_binary_path();
     int fd = open(bin_name, O_RDONLY);
+    free(bin_name);
```

Comment:
For next time, I think there's a better way than mmaping the binary in. Don't worry about it for now though.

Reply:
Out of curiosity, how would you have done this?

Reply:
You could read in just the header using a regular `read`/`fread` and cast it to a Elf header. You could even meet me at the middle and mmap in only the header (rather than the whole file).

It doesn't really matter tbh.

Reply:
I see, I'll keep that in mind.